boot/freeldr/freeldr/rosload.cmake

```diff
@@ -0,0 +1,104 @@
+##
+## PROJECT:     FreeLoader
+## LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+## PURPOSE:     Build definitions for rosload 2nd stage loader
+## COPYRIGHT:   Copyright 2024 Timo Kreuzer <timo.kreuzer@reactos.org>
+##
+
+#spec2def(rosload.exe rosload.spec)
+
+list(APPEND ROSLOAD_SOURCE
+    include/freeldr.h
+    bootmgr.c
+    custom.c
+    linuxboot.c
+    miscboot.c
+    options.c
+    oslist.c
```

Comment:
See my previous comment: All these doesn't need to be in rosload; this shouldn't be its purpose.
The purpose of rosload.exe would be to just start ReactOS (or NT-type of OSes).
The main boot menu and these custom non-ReactOS/NT things should stay in the freeldr.sys.

Reply:
All these don't need to be in freeldr.sys. And all these contain dependecies on ntldr code. And I disagree on keeping anything in freeldr, which we don't need there. Also untangling the dependency mess just to conform to some arbitrary feelings of what should go where would be a PITA and absolutely not worth it.